src/multitask_personalization/rom/models.py

```diff
@@ -272,27 +312,25 @@ def __init__(
                 dim_max - dim_min
             )
         self._dense_joint_samples = joint_angle_samples
-        self.update_parameters(self._rom_model_context_parameters)
+        self.set_trainable_parameters(self._rom_model_context_parameters)
         print("Learned ROM model created")
 
         # Uncomment for debugging.
         # self._visualize_reachable_points()
 
-    def get_parameter_size(self) -> int:
-        """Get the size of the parameter."""
-        return self._parameter_size
+    def check_position_reachable(self, position: NDArray) -> bool:
+        distance, _ = self._reachable_kd_tree.query(position)
+        return distance < self._ik_distance_threshold
 
-    def get_rom_model_context_parameters(self) -> NDArray:
-        """Get the ROM model context parameters."""
-        return self._rom_model_context_parameters
+    def sample_reachable_position(self, rng: np.random.Generator) -> NDArray:
+        return rng.choice(self._reachable_points)
 
-    def get_reachable_points(self) -> list[NDArray]:
-        """Get the reachable points."""
-        return self._reachable_points
+    def get_trainable_parameters(self) -> Any:
+        return self._rom_model_context_parameters.copy()
 
-    def update_parameters(self, parameters: NDArray) -> None:
-        """Update the ROM model parameters."""
-        self._rom_model_context_parameters = parameters
+    def set_trainable_parameters(self, params: Any) -> None:
+        assert isinstance(params, np.ndarray)
+        self._rom_model_context_parameters = params
```

Comment:
Is making a copy of params needed here?

Reply:
oh wait sorry I misread your comment. yes good idea!